### PR TITLE
[POC] Edit dashboard

### PR DESCRIPTION
**Pull request by quanvo0213**
Please add in Config file
```



from flask_appbuilder.security.views import expose
from superset.security import SupersetSecurityManager
from flask_appbuilder.security.manager import BaseSecurityManager
from flask_appbuilder.security.manager import AUTH_REMOTE_USER
from flask import  redirect, request, flash
from flask_login import login_user

# Create a custom view to authenticate the user
AuthRemoteUserView=BaseSecurityManager.authremoteuserview
class CustomAuthUserView(AuthRemoteUserView):
    @expose('/login/')
    def login(self):
        token = request.args.get('token')
        next = request.args.get('next')
        isEmbeddedKatalonMode = request.args.get('isKatalonEmbeddedMode')
        projectId = request.args.get('projectId')
        accessToken = request.args.get('accessToken')
        sm = self.appbuilder.sm
        session = sm.get_session
        user = session.query(sm.user_model).filter_by(username='admin').first()
        if token == '1234abcd456':
            login_user(user, remember=False, force=True)
            if (next is not None):
                return redirect(next + '&isKatalonEmbeddedMode=' + isEmbeddedKatalonMode + '&projectId=' + projectId + '&accessToken=' + accessToken)
            else:
                return redirect(self.appbuilder.get_url_for_index)
        else:
            flash('Unable to auto login', 'warning')
            return super(CustomAuthUserView,self).login()

# Create a custom Security manager that overrides the CustomAuthUserView
class CustomSecurityManager(SupersetSecurityManager):
    authremoteuserview = CustomAuthUserView

# Use our custom authenticator
CUSTOM_SECURITY_MANAGER = CustomSecurityManager

# User remote authentication
AUTH_TYPE = AUTH_REMOTE_USER
```

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/connection/SupersetClientClass.ts [hunk 1]
```diff
@@ -88,8 +88,10 @@ export default class SupersetClientClass {
       // as the base of baseUrl
       window.location.href,
     );
-    this.baseUrl = url.href.replace(/\/+$/, ''); // always strip trailing slash
-    this.host = url.host;
+    // this.baseUrl = url.href.replace(/\/+$/, ''); // always strip trailing slash
+    this.baseUrl = 'http://localhost:8080/proxy';
+    // this.host = url.host;
+    this.host = 'localhost:8080/proxy';
     this.protocol = url.protocol as Protocol;
```
Comment (quanvo0213) [inline superset-frontend/packages/superset-ui-core/src/connection/SupersetClientClass.ts]: using env var

File: superset-frontend/packages/superset-ui-core/src/connection/SupersetClientClass.ts [hunk 2]
```diff
@@ -198,13 +200,24 @@ export default class SupersetClientClass {
...
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const projectId = urlParams.get('projectId');
+    const accessToken = urlParams.get('accessToken');
+    const combinedHeaders = {
+      ...this.headers,
+      ...headers,
+      ...(projectId && { 'x-project-id': projectId }),
+      ...(accessToken && { Authorization: `Bearer ${accessToken}` }),
+    };
```
Comment (quanvo0213) [inline superset-frontend/packages/superset-ui-core/src/connection/SupersetClientClass.ts:214]: create new class in Katalon folder

File: superset-frontend/packages/superset-ui-core/src/connection/callApi/callApi.ts [hunk 3]
```diff
@@ -66,7 +66,7 @@ export default async function callApi({
   fetchRetryOptions,
   headers,
   method = 'GET',
-  mode = 'same-origin',
+  mode = 'cors',
   postPayload,
```
Comment (quanvo0213) [inline superset-frontend/packages/superset-ui-core/src/connection/callApi/callApi.ts:70]: create new class in Katalon folder

File: superset-frontend/src/components/Chart/chartAction.js [hunk 4]
```diff
@@ -189,7 +189,8 @@ const v1ChartDataRequest = async (
     // eslint-disable-next-line camelcase
     domainShardingEnabled && requestParams?.dashboard_id;
   const url = getChartDataUri({
-    path: '/api/v1/chart/data',
```
Comment (quanvo0213) [inline superset-frontend/src/components/Chart/chartAction.js]: create new class in Katalon folder

File: superset-frontend/src/constants.ts [hunk 5]
```diff
@@ -111,6 +111,18 @@ export const URL_PARAMS = {
...
+    type: 'boolean',
+  },
+  projectId: {
+    name: 'projectId',
+    type: 'string',
+  },
+  accessToken: {
+    name: 'accessToken',
+    type: 'string',
+  },
```
Comment (quanvo0213) [inline superset-frontend/src/constants.ts:125]: if not using proxy, remove

File: superset-frontend/src/dashboard/actions/dashboardState.js [hunk 6]
```diff
@@ -342,11 +344,20 @@ export function saveDashboardRequest(data, id, saveType) {
...
+      const accessToken = getUrlParam(URL_PARAMS.accessToken);
+
       // redirect to the new slug or id
       window.history.pushState(
         { event: 'dashboard_properties_changed' },
         '',
-        `/superset/dashboard/${slug || id}/`,
+        `/superset/dashboard/${
+          slug || id
+        }/?isKatalonEmbeddedMode=${isKatalonEmbeddedDashboard}&projectId=${projectId}&accessToken=${accessToken}`,
```
Comment (quanvo0213) [inline superset-frontend/src/dashboard/actions/dashboardState.js:360]: remove

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 7]
```diff
@@ -277,7 +279,15 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
       // Go to new dashboard url
       if (gotodash && dashboard) {
-        this.props.history.push(dashboard.url);
+        const isKatalonEmbeddedMode = getUrlParam(
+          URL_PARAMS.isKatalonEmbeddedMode,
+        );
+        let { url } = dashboard;
+        if (isKatalonEmbeddedMode) {
+          url += '?standalone=1&isKatalonEmbeddedMode=true';
+        }
```
Comment (quanvo0213) [inline superset-frontend/src/explore/components/SaveModal.tsx:288]: removing when still not using iframe

File: superset-frontend/src/pages/Chart/index.tsx [hunk 8]
```diff
@@ -146,7 +146,13 @@ export default function ExplorePage() {
...
         .catch(err => {
           dispatch(hydrateExplore(fallbackExploreInitialData));
-          dispatch(addDangerToast(err.message));
+
+          const isKatalonEmbeddedMode = getUrlParam(
+            URL_PARAMS.isKatalonEmbeddedMode,
+          );
+          if (!isKatalonEmbeddedMode) {
+            dispatch(addDangerToast(err.message));
+          }
```
Comment (quanvo0213) [inline superset-frontend/src/pages/Chart/index.tsx:155]: .

File: superset-frontend/src/features/home/Menu.tsx [hunk 9]
```diff
@@ -238,6 +238,9 @@ export function Menu({
 
   const standalone = getUrlParam(URL_PARAMS.standalone);
   if (standalone || uiConfig.hideNav) return <></>;
+  if (getUrlParam(URL_PARAMS.isKatalonEmbeddedMode)) {
```
Comment (quanvo0213) [inline superset-frontend/src/features/home/Menu.tsx:241]: .

File: superset-frontend/src/explore/exploreUtils/index.js [hunk 10]
```diff
@@ -111,7 +111,8 @@ export function getChartDataUri({ path, qs, allowDomainSharding = false }) {
   let uri = new URI({
     protocol: window.location.protocol.slice(0, -1),
     hostname: getHostName(allowDomainSharding),
-    port: window.location.port ? window.location.port : '',
+    // port: window.location.port ? window.location.port : '',
+    port: '8080',
```
Comment (quanvo0213) [inline superset-frontend/src/explore/exploreUtils/index.js]: fix for deployment

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 11]
```diff
@@ -392,8 +392,12 @@ class DatasourceControl extends React.PureComponent {
       }
     }
 
-    const titleText = isMissingDatasource
+    const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
+    const text = !isKatalonEmbeddedMode
       ? t('Missing dataset')
```
Comment (quanvo0213) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:397]: .

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 12]
```diff
@@ -443,43 +459,49 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
-            !this.state.datasetName)
-        }
-        data-test="btn-modal-save"
-      >
-        {t('Save')}
-      </Button>
-    </div>
-  );
+  renderFooter = () => {
+    const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
```
Comment (quanvo0213) [inline superset-frontend/src/explore/components/SaveModal.tsx:463]: .

File: superset-frontend/src/explore/components/ExploreViewContainer/index.jsx [hunk 13]
```diff
@@ -699,7 +701,11 @@ function ExploreViewContainer(props) {
           actions={props.actions}
           form_data={props.form_data}
           sliceName={props.sliceName}
-          dashboardId={props.dashboardId}
+          dashboardId={
+            !isKatalonEmbeddedMode
+              ? props.dashboardId
```
Comment (quanvo0213) [inline superset-frontend/src/explore/components/ExploreViewContainer/index.jsx:706]: .

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 14]
```diff
@@ -741,36 +743,48 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
...
-            expandIconPosition="right"
-            ghost
-          >
-            {showDatasourceAlert && <DatasourceAlert />}
-            {querySections.map(renderControlPanelSection)}
-          </Collapse>
-        </Tabs.TabPane>
-        {showCustomizeTab && (
-          <Tabs.TabPane key="display" tab={t('Customize')}>
+      {!isKatalonEmbeddedMode ? (
```
Comment (quanvo0213) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:750]: .

File: superset-frontend/src/explore/actions/saveModalActions.js [hunk 15]
```diff
@@ -20,6 +20,8 @@ import rison from 'rison';
 import { SupersetClient, t } from '@superset-ui/core';
 import { addSuccessToast } from 'src/components/MessageToasts/actions';
 import { isEmpty } from 'lodash';
```
Comment (quanvo0213) [inline superset-frontend/src/explore/actions/saveModalActions.js:22]: remove unuse code

File: superset-frontend/src/dashboard/components/SliceHeader/index.tsx [hunk 16]
```diff
@@ -258,10 +262,10 @@ const SliceHeader: FC<SliceHeaderProps> = ({
...
               </Tooltip>
             )}
-            {!uiConfig.hideChartControls && (
+            {!isKatalonEmbeddedMode && !uiConfig.hideChartControls && (
               <FiltersBadge chartId={slice.slice_id} />
             )}
-            {!uiConfig.hideChartControls && (
+            {!isKatalonEmbeddedMode && !uiConfig.hideChartControls && (
               <SliceHeaderControls
                 slice={slice}
```
Comment (quanvo0213) [inline superset-frontend/src/dashboard/components/SliceHeader/index.tsx:270]: .

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 17]
```diff
@@ -331,13 +333,25 @@ class SliceAdder extends React.Component {
...
+            onClick={() => {
+              const isKatalonEmbeddedDashboard = getUrlParam(
+                URL_PARAMS.isKatalonEmbeddedMode,
+              );
+              const projectId = getUrlParam(URL_PARAMS.projectId);
+              const accessToken = getUrlParam(URL_PARAMS.accessToken);
+
+              if (isKatalonEmbeddedDashboard) {
+                window.location.assign(
+                  `/explore/?dashboard_id=${this.props.dashboardId}&isKatalonEmbeddedMode=true&projectId=${projectId}&accessToken=${accessToken}`,
```
Comment (quanvo0213) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:345]: .

File: superset-frontend/src/dashboard/components/Header/index.jsx [hunk 18]
```diff
@@ -466,10 +467,12 @@ class Header extends React.PureComponent {
...
 
-    const userCanEdit =
-      dashboardInfo.dash_edit_perm && !dashboardInfo.is_managed_externally;
+    // const userCanEdit =
+    //   dashboardInfo.dash_edit_perm && !dashboardInfo.is_managed_externally;
+    const userCanEdit = true;
     const userCanShare = dashboardInfo.dash_share_perm;
-    const userCanSaveAs = dashboardInfo.dash_save_perm;
+    // const userCanSaveAs = dashboardInfo.dash_save_perm;
+    const userCanSaveAs = true;
```
Comment (quanvo0213) [inline superset-frontend/src/dashboard/components/Header/index.jsx:475]: .

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx [hunk 19]
```diff
@@ -134,24 +136,31 @@ export const PageHeaderWithActions = ({
   tooltipProps,
 }: PageHeaderWithActionsProps) => {
   const theme = useTheme();
+  const isKatalonEmbeddedDashboard = getUrlParam(
+    URL_PARAMS.isKatalonEmbeddedMode,
+  );
```
Comment (huynguyen2908) [inline superset-frontend/src/components/PageHeaderWithActions/index.tsx:141]: Move this to a util.

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx [hunk 20]
```diff
@@ -134,24 +136,31 @@ export const PageHeaderWithActions = ({
...
+          <>
+            <DynamicEditableTitle {...editableTitleProps} />
+            {showTitlePanelItems && (
+              <div css={buttonsStyles}>
+                {certificatiedBadgeProps?.certifiedBy && (
+                  <CertifiedBadge {...certificatiedBadgeProps} />
+                )}
+                {showFaveStar && <FaveStar {...faveStarProps} />}
+                {titlePanelAdditionalItems}
+              </div>
```
Comment (huynguyen2908) [inline superset-frontend/src/components/PageHeaderWithActions/index.tsx:155]: Conflict.

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx [hunk 21]
```diff
@@ -134,24 +136,31 @@ export const PageHeaderWithActions = ({
...
         )}
       </div>
       <div className="right-button-panel">
         {rightPanelAdditionalItems}
         <div css={additionalActionsContainerStyles}>
-          {showMenuDropdown && (
+          {!isKatalonEmbeddedDashboard && showMenuDropdown && (
             <AntdDropdown
               trigger={['click']}
               overlay={additionalActionsMenu}
```
Comment (huynguyen2908) [inline superset-frontend/src/components/PageHeaderWithActions/index.tsx:166]: Conflict.

File: superset-frontend/src/dashboard/actions/dashboardState.js [hunk 22]
```diff
@@ -342,11 +344,20 @@ export function saveDashboardRequest(data, id, saveType) {
         dispatch(saveDashboardRequestSuccess(lastModifiedTime));
       }
       dispatch(saveDashboardFinished());
+
+      const isKatalonEmbeddedDashboard = getUrlParam(
+        URL_PARAMS.isKatalonEmbeddedMode,
+      );
+      const projectId = getUrlParam(URL_PARAMS.projectId);
+      const accessToken = getUrlParam(URL_PARAMS.accessToken);
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/actions/dashboardState.js:352]: Conflict.

File: superset-frontend/src/dashboard/actions/dashboardState.js (hunk 6 above)
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/actions/dashboardState.js:360]: Conflict.

File: superset-frontend/src/dashboard/components/Header/index.jsx (hunk 18 above)
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/Header/index.jsx:475]: Conflict.

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 23]
```diff
@@ -45,6 +45,8 @@ import { slicePropShape } from 'src/dashboard/util/propShapes';
 import { debounce, pickBy } from 'lodash';
 import Checkbox from 'src/components/Checkbox';
 import { InfoTooltipWithTrigger } from '@superset-ui/chart-controls';
+import { getUrlParam } from 'src/utils/urlUtils';
+import { URL_PARAMS } from 'src/constants';
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:49]: Conflict

File: superset-frontend/src/dashboard/components/SliceAdder.jsx (hunk 17 above)
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:345]: Conflict

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 24]
```diff
@@ -331,13 +333,25 @@ class SliceAdder extends React.Component {
...
+                  `/explore/?dashboard_id=${this.props.dashboardId}&isKatalonEmbeddedMode=true&projectId=${projectId}&accessToken=${accessToken}`,
+                );
+              } else {
+                window.open(
+                  `/chart/add?dashboard_id=${this.props.dashboardId}`,
+                  '_blank',
+                  'noopener noreferrer',
+                );
+              }
+            }}
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:354]: Conflict.

File: superset-frontend/src/dashboard/components/SliceHeader/index.tsx [hunk 25]
```diff
@@ -37,6 +37,8 @@ import Icons from 'src/components/Icons';
 import { RootState } from 'src/dashboard/types';
 import { getSliceHeaderTooltip } from 'src/dashboard/util/getSliceHeaderTooltip';
 import { DashboardPageIdContext } from 'src/dashboard/containers/DashboardPage';
+import { getUrlParam } from 'src/utils/urlUtils';
+import { URL_PARAMS } from 'src/constants';
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceHeader/index.tsx:41]: Conflict

File: superset-frontend/src/dashboard/components/SliceHeader/index.tsx [hunk 26]
```diff
@@ -195,6 +197,8 @@ const SliceHeader: FC<SliceHeaderProps> = ({
 
   const exploreUrl = `/explore/?dashboard_page_id=${dashboardPageId}&slice_id=${slice.slice_id}`;
 
+  const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceHeader/index.tsx:200]: Conflict

File: superset-frontend/src/dashboard/components/SliceHeader/index.tsx [hunk 27]
```diff
@@ -258,10 +262,10 @@ const SliceHeader: FC<SliceHeaderProps> = ({
                 <CrossFilterIcon iconSize="m" />
               </Tooltip>
             )}
-            {!uiConfig.hideChartControls && (
+            {!isKatalonEmbeddedMode && !uiConfig.hideChartControls && (
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceHeader/index.tsx:265]: Conflict

File: superset-frontend/src/dashboard/components/SliceHeader/index.tsx [hunk 28]
```diff
@@ -258,10 +262,10 @@ const SliceHeader: FC<SliceHeaderProps> = ({
...
               <FiltersBadge chartId={slice.slice_id} />
             )}
-            {!uiConfig.hideChartControls && (
+            {!isKatalonEmbeddedMode && !uiConfig.hideChartControls && (
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/SliceHeader/index.tsx:268]: Conflict

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx [hunk 29]
```diff
@@ -56,6 +56,7 @@ import DropdownContainer, {
   Ref as DropdownContainerRef,
 } from 'src/components/DropdownContainer';
 import Icons from 'src/components/Icons';
+import FilterExtension from 'src/katalon/CustomFilter/FilterExtension';
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx:59]: Conflict

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx [hunk 30]
```diff
@@ -215,6 +216,7 @@ const FilterControls: FC<FilterControlsProps> = ({
     if (isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS)) {
       const nativeFiltersInScope = filtersInScope.map((filter, index) => ({
         id: filter.id,
+        name: filter.name,
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx:219]: Conflict.

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx [hunk 31]
```diff
@@ -316,6 +318,8 @@ const FilterControls: FC<FilterControlsProps> = ({
     }
   }, [outlinedFilterId, lastUpdated, popoverRef, overflowedIds]);
 
+  const renderFilterExtension = () => <FilterExtension items={items} />;
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx:321]: Conflict.

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx [hunk 32]
```diff
@@ -332,7 +336,9 @@ const FilterControls: FC<FilterControlsProps> = ({
       {filterBarOrientation === FilterBarOrientation.VERTICAL &&
         renderVerticalContent()}
       {filterBarOrientation === FilterBarOrientation.HORIZONTAL &&
-        renderHorizontalContent()}
+        renderFilterExtension()}
+      {/* {filterBarOrientation === FilterBarOrientation.HORIZONTAL &&
+        renderHorizontalContent()} */}
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterControls.tsx:341]: Conflict.

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx [hunk 33]
```diff
@@ -48,6 +48,7 @@ import { logEvent } from 'src/logger/actions';
 import { LOG_ACTIONS_CHANGE_DASHBOARD_FILTER } from 'src/logger/LogUtils';
 import { FilterBarOrientation, RootState } from 'src/dashboard/types';
 import { UserWithPermissionsAndRoles } from 'src/types/bootstrapTypes';
+import KatalonHorizontal from 'src/katalon/CustomFilter/KatalonHorizontal';
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx:51]: Conflict

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx [hunk 34]
```diff
@@ -287,7 +288,7 @@ const FilterBar: React.FC<FiltersBarProps> = ({
 
   const filterBarComponent =
     orientation === FilterBarOrientation.HORIZONTAL ? (
-      <Horizontal
+      <KatalonHorizontal
```
Comment (huynguyen2908) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx:291]: Conflict

File: superset-frontend/src/explore/actions/saveModalActions.js [hunk 35]
```diff
@@ -20,6 +20,8 @@ import rison from 'rison';
...
+import { getUrlParam } from 'src/utils/urlUtils';
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/actions/saveModalActions.js:23]: Conflict

File: superset-frontend/src/explore/actions/saveModalActions.js [hunk 36]
```diff
@@ -20,6 +20,8 @@ import rison from 'rison';
...
+import { URL_PARAMS } from 'src/constants';
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/actions/saveModalActions.js:24]: Conflict

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 37]
```diff
@@ -65,6 +65,8 @@ import { ExploreActions } from 'src/explore/actions/exploreActions';
 import { ChartState, ExplorePageState } from 'src/explore/types';
 import { Tooltip } from 'src/components/Tooltip';
 import Icons from 'src/components/Icons';
+import { getUrlParam } from 'src/utils/urlUtils';
+import { URL_PARAMS } from 'src/constants';
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:69]: Conflict

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 38]
```diff
@@ -741,36 +743,48 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
 
   const showCustomizeTab = customizeSections.length > 0;
 
+  const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:746]: Conflict

File: superset-frontend/src/explore/components/ControlPanelsContainer.tsx [hunk 39]
```diff
@@ -741,36 +743,48 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
...
+        </ControlPanelsTabs>
+      ) : (
+        <Collapse
+          defaultActiveKey={expandedQuerySections}
+          expandIconPosition="right"
+          ghost
+        >
+          {querySections.map(renderControlPanelSection)}
+        </Collapse>
+      )}
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/ControlPanelsContainer.tsx:787]: Conflict.

File: superset-frontend/src/explore/components/ExploreViewContainer/index.jsx [hunk 40]
```diff
@@ -553,6 +553,8 @@ function ExploreViewContainer(props) {
     return renderChartContainer();
   }
 
+  const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/ExploreViewContainer/index.jsx:556]: Conflict.

File: superset-frontend/src/explore/components/ExploreViewContainer/index.jsx [hunk 41]
```diff
@@ -699,7 +701,11 @@ function ExploreViewContainer(props) {
           actions={props.actions}
           form_data={props.form_data}
           sliceName={props.sliceName}
-          dashboardId={props.dashboardId}
+          dashboardId={
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/ExploreViewContainer/index.jsx:704]: Conflict.

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 42]
```diff
@@ -46,6 +46,8 @@ import { setSaveChartModalVisibility } from 'src/explore/actions/saveModalAction
 import { SaveActionType } from 'src/explore/types';
 import { UserWithPermissionsAndRoles } from 'src/types/bootstrapTypes';
 import { Dashboard } from 'src/types/Dashboard';
+import { getUrlParam } from 'src/utils/urlUtils';
+import { URL_PARAMS } from 'src/constants';
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:51]: Conflict

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 43]
```diff
@@ -277,7 +279,15 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
-        this.props.history.push(dashboard.url);
+        const isKatalonEmbeddedMode = getUrlParam(
+          URL_PARAMS.isKatalonEmbeddedMode,
+        );
+        let { url } = dashboard;
+        if (isKatalonEmbeddedMode) {
+          url += '?standalone=1&isKatalonEmbeddedMode=true';
+        }
+
+        this.props.history.push(url);
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:290]: Conflict

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 44]
```diff
@@ -335,28 +345,33 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
 
   renderSaveChartModal = () => {
     const info = this.info();
+    const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:348]: Conflict.

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 45]
```diff
@@ -335,28 +345,33 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
+                data-test="saveas-radio"
+                checked={this.state.action === 'saveas'}
+                onChange={() => this.changeAction('saveas')}
+              >
+                {t('Save as...')}
+              </Radio>
+            </FormItem>
+            <hr />
+          </>
+        )}
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:374]: Conflict.

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 46]
```diff
@@ -384,6 +399,7 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
           </FormItem>
         )}
         {!(
+          isKatalonEmbeddedMode &&
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:402]: Conflict.

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 47]
```diff
@@ -443,43 +459,49 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
...
+                !this.state.datasetName)
+            }
+            data-test="btn-modal-save"
+          >
+            {t('Save')}
+          </Button>
+        )}
+      </div>
+    );
+  };
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/SaveModal.tsx:504]: Conflict.

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 48]
```diff
@@ -392,8 +392,12 @@ class DatasourceControl extends React.PureComponent {
       }
     }
 
-    const titleText = isMissingDatasource
+    const isKatalonEmbeddedMode = getUrlParam(URL_PARAMS.isKatalonEmbeddedMode);
+    const text = !isKatalonEmbeddedMode
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:396]: Conflict

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 49]
```diff
@@ -427,7 +431,7 @@ class DatasourceControl extends React.PureComponent {
           </AntdDropdown>
         </div>
         {/* missing dataset */}
-        {isMissingDatasource && isMissingParams && (
+        {!isKatalonEmbeddedMode && isMissingDatasource && isMissingParams && (
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:434]: Conflict.

File: superset-frontend/src/explore/exploreUtils/index.js [hunk 50]
```diff
@@ -111,8 +112,10 @@ export function getChartDataUri({ path, qs, allowDomainSharding = false }) {
   let uri = new URI({
     protocol: window.location.protocol.slice(0, -1),
     hostname: getHostName(allowDomainSharding),
-    port: window.location.port ? window.location.port : '',
-    path,
+    // port: window.location.port ? window.location.port : '',
+    port: Config.chartConfig.port,
+    // path,
+    path: Config.chartConfig.path + path,
```
Comment (huynguyen2908) [inline superset-frontend/src/explore/exploreUtils/index.js:118]: Conflict.

File: superset-frontend/src/features/home/Menu.tsx [hunk 51]
```diff
@@ -238,6 +238,9 @@ export function Menu({
...
+    return <></>;
+  }
```
Comment (huynguyen2908) [inline superset-frontend/src/features/home/Menu.tsx:243]: Conflict.

File: superset-frontend/src/pages/Chart/index.tsx (hunk 8 above)
Comment (huynguyen2908) [inline superset-frontend/src/pages/Chart/index.tsx:155]: Conflict.

File: superset-frontend/src/utils/hostNamesConfig.js [hunk 52]
```diff
@@ -18,14 +18,16 @@
  */
 import { initFeatureFlags } from '@superset-ui/core';
 import getBootstrapData from './getBootstrapData';
+import Config from '../../config';
```
Comment (huynguyen2908) [inline superset-frontend/src/utils/hostNamesConfig.js:21]: Conflict.

File: superset-frontend/src/utils/hostNamesConfig.js [hunk 53]
```diff
@@ -18,14 +18,16 @@
...
 
 function getDomainsConfig() {
   const appContainer = document.getElementById('app');
   if (!appContainer) {
     return [];
   }
 
-  const availableDomains = new Set([window.location.hostname]);
+  // const availableDomains = new Set([window.location.hostname]);
+  const availableDomains = new Set([Config.chartConfig.host]);
```
Comment (huynguyen2908) [inline superset-frontend/src/utils/hostNamesConfig.js:30]: Conflict

File: superset-frontend/src/visualizations/presets/MainPreset.js [hunk 54]
```diff
@@ -76,6 +76,10 @@ import {
   TimeGrainFilterPlugin,
   GroupByFilterPlugin,
 } from 'src/filters/components';
+import {
+  KatalonTimeFilterPlugin,
+  KatalonSelectFilterPlugin
+} from 'src/katalon/filters/components';
```
Comment (huynguyen2908) [inline superset-frontend/src/visualizations/presets/MainPreset.js:82]: Conflict

File: superset-frontend/src/visualizations/presets/MainPreset.js [hunk 55]
```diff
@@ -157,9 +161,11 @@ export default class MainPreset extends Preset {
         new EchartsWaterfallChartPlugin().configure({
           key: 'waterfall',
         }),
-        new SelectFilterPlugin().configure({ key: 'filter_select' }),
+        // new SelectFilterPlugin().configure({ key: 'filter_select' }),
+        new KatalonSelectFilterPlugin().configure({ key: 'filter_select' }),
         new RangeFilterPlugin().configure({ key: 'filter_range' }),
-        new TimeFilterPlugin().configure({ key: 'filter_time' }),
+        // new TimeFilterPlugin().configure({ key: 'filter_time' }),
+        new KatalonTimeFilterPlugin().configure({ key: 'filter_time' }),
```
Comment (huynguyen2908) [inline superset-frontend/src/visualizations/presets/MainPreset.js:168]: Conflict

**Pull request merged**